endpoint.go

```diff
@@ -133,6 +199,37 @@ func (ep *endpoint) Network() string {
 	return ep.network.name
 }
 
+func (ep *endpoint) Key() []string {
+	return []string{datastore.EndpointKeyPrefix, string(ep.network.id), string(ep.id)}
+}
+
+func (ep *endpoint) KeyPrefix() []string {
+	return []string{datastore.EndpointKeyPrefix, string(ep.network.id)}
+}
+
+func (ep *endpoint) networkIDFromKey(key []string) (types.UUID, error) {
+	if key == nil || len(key) < 3 || key[0] != datastore.EndpointKeyPrefix {
+		return types.UUID(""), fmt.Errorf("Invalid endpoint key : %v", key)
+	}
+	return types.UUID(key[1]), nil
```

Comment:
We need to get some structure to the key positions here. What does `key[1]` mean? If not may be a comment on the key structure would go a long way